dspace-api/src/main/java/org/dspace/app/mediafilter/TikaTextExtractionFilter.java

```diff
@@ -0,0 +1,178 @@
+import org.apache.tika.sax.BodyContentHandler;
+import org.apache.tika.sax.ContentHandlerDecorator;
+import org.dspace.content.Item;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.xml.sax.SAXException;
+
+/**
+ * Text Extraction media filter which uses Apache Tika to extract text from a large number of file formats (including
+ * all Microsoft formats, PDF, HTML, Text, etc).  For a more complete list of file formats supported by Tika see the
+ * Tika documentation: https://tika.apache.org/2.3.0/formats.html
+ */
+public class TikaTextExtractionFilter
+    extends MediaFilter {
+    private final static Logger log = LogManager.getLogger();
+
+    @Override
+    public String getFilteredName(String oldFilename) {
+        return oldFilename + ".txt";
+    }
+
+    @Override
+    public String getBundleName() {
+        return "TEXT";
+    }
+
+    @Override
+    public String getFormatString() {
+        return "Text";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Extracted text";
+    }
+
+    @Override
+    public InputStream getDestinationStream(Item currentItem, InputStream source, boolean verbose)
+        throws Exception {
+        ConfigurationService configurationService = DSpaceServicesFactory.getInstance().getConfigurationService();
+        boolean useTemporaryFile = configurationService.getBooleanProperty("textextractor.use-temp-file", false);
+
+        if (useTemporaryFile) {
+            // Extract text out of source file using a temp file, returning results as InputStream
+            return extractUsingTempFile(source, verbose);
+        }
+
+        // Not using temporary file. We'll use Tika's default in-memory parsing.
+        // Get maximum characters to extract. Default is 100,000 chars, which is also Tika's default setting.
+        String extractedText;
+        int maxChars = configurationService.getIntProperty("textextractor.max-chars", 100000);
+        try {
+            // Use Tika to extract text from input. Tika will automatically detect the file type.
+            Tika tika = new Tika();
+            tika.setMaxStringLength(maxChars); // Tell Tika the maximum number of characters to extract
+            extractedText = tika.parseToString(source);
+        } catch (IOException e) {
+            System.err.println("Unable to extract text from bitstream in Item " + currentItem.getID().toString());
+            e.printStackTrace();
+            log.error("Unable to extract text from bitstream in Item {}", currentItem.getID().toString(), e);
+            throw e;
+        } catch (OutOfMemoryError oe) {
+            System.err.println("OutOfMemoryError occurred when extracting text from bitstream in Item " +
+                                   currentItem.getID().toString() +
+                                   "You may wish to enable 'textextractor.use-temp-file'.");
```

Comment:
Needs space between UUID and start of second sentence.

You may wish to use `System.err.format("blabla in Item %s.  You may blabla%n", currentItem.getID().toString());` to make it easier to see the spacing and make this more like the logger call.

Reply:
@mwoodiupui : Good catch. Yep, you are right that using `System.err.format` would probably be better here.  I just pushed up a commit with that minor change.